client/settings/check-starcraft-path.js

```diff
@@ -42,14 +58,22 @@ async function checkHash(path, validHashes) {
 // be checked for other copies. If downgradePath contains files that match the correct hashes, this
 // will be counted as having the correct version, but `downgradePath` will be true.
 export async function checkStarcraftPath(dirPath, downgradePath) {
-  const requiredFiles = ['starcraft.exe', 'storm.dll', 'local.dll', 'stardat.mpq', 'broodat.mpq']
+  const requiredFiles = ['starcraft.exe', 'storm.dll', 'stardat.mpq', 'broodat.mpq']
 
   try {
     await Promise.all(requiredFiles.map(f => accessAsync(path.join(dirPath, f), fs.constants.R_OK)))
   } catch (err) {
     return { path: false, version: false, downgradePath: false }
   }
 
+  // Due to 1.19 version moving local.dll to a separate folder, we need to handle it separately
+  const matches = await globAsync(`${dirPath}/**/local.dll`)
+  if (matches.length > 0) {
+    maybeCopyLocalDll(matches[0], downgradePath)
```

Comment:
You're performing a filesystem mutation in what was previously pure, this isn't good. On top of that, I don't think we should be copying this unless we need it (that is, unless the file is in the wrong spot, or the other files require us to downgrade). Thus, this method should be updated to simply check for local.dll and mark the version as wrong if it's not found (as an added check to the existing ones here), and the file that deals with downgrading should actually perform any necessary copying.